src/caselawclient/xml_tools.py

```diff
@@ -3,26 +3,30 @@
 
 akn_uk_namespaces = {"akn": "http://docs.oasis-open.org/legaldocml/ns/akn/3.0",
                      "uk": "https://caselaw.nationalarchives.gov.uk/akn"}
+akn_namespace_uri = "http://docs.oasis-open.org/legaldocml/ns/akn/3.0"
+uk_namespace_uri = "https://caselaw.nationalarchives.gov.uk/akn"
 search_namespace = {"search": "http://marklogic.com/appservices/search"}
 
 
-class JudgmentMissingMetadataError(IndexError):
-    pass
-
-
 def get_metadata_name_value(xml: ElementTree) -> str:
     name = get_metadata_name_element(xml)
-    return name.attrib["value"]
+    value = name.attrib["value"]
+    if value is None:
+        return ""
+    return value
 
 
-def get_element(xml: ElementTree, xpath) -> Element:
+def get_element(xml: ElementTree, xpath, element_name="FRBRname", element_namespace=akn_namespace_uri, has_value_attribute=True) -> Element:
     name = xml.find(
         xpath,
         namespaces=akn_uk_namespaces,
     )
 
     if name is None:
-        raise JudgmentMissingMetadataError
+        element = ElementTree.Element(ElementTree.QName(element_namespace, element_name))
```

Comment:
I see that this creates an element if the element doesn't exist, but is it created at the right point in the tree? If the value of this is set, will it actually be saved back into the XML document properly? Feels like it's just a floating element, unattached to `xml`?

Reply:
Right, this is definitely the case *but* we can find the element's parent in the editor UI project and insert it into the parent. This works! (Tried it!). It's not very elegant though.